tools/python/CMakeLists.txt

```diff
@@ -13,35 +13,10 @@ endif()
 
 project(dlib_python_bindings)
 
-# Pybind11's cmake scripts enable link time optimization by default.  However,
-# it makes linking take a really long time and doesn't seem to substantively
-# improve runtime performance.  So we disable LTO here to make building dlib
-# faster.
-set(PYBIND11_LTO_CXX_FLAGS "")
```

Comment:
Totally keep this too.  It takes like 3 minutes to link otherwise on a fast machine with a recent version of gcc.  I forget if this sets anything in visual studio, but visual studio is also *massively* slow.  People already post a lot about the python bindings taking a long time to compile.  Which they don't really on a normal system with reasonable compilers, but lots of people use windows, where compilers are terribly slow, or they compile on a raspberry pi or some very low powered machine.

Reply:
I just tested this and it doesn't seem to do anything anymore. At some point one of the pybind11 updates must have made this not disable LTO anymore :(   I need to fix that, people complain about slow python binding build times too much.

Reply:
We can disable LTO more cleanly using:
```
set_target_properties(_dlib_pybind11 PROPERTIES
   INTERPROCEDURAL_OPTIMIZATION OFF)
```

Reply:
Sweet

I mean I use it normally. It's just that the python bindings need to instantiate a huge number of templates so they can all be bound into python. And that ends up taking a while. Like in c++ you wouldn't normally do that, but python obviously has no concept of templates or compiling code so anything anyone might want needs to be all compiled when making the binding. And there are a ton of variants and options in doing to bind.